services/121-service/src/payments/fsp-integration/intersolve-visa/intersolve-visa.api.service.ts

```diff
@@ -66,50 +91,131 @@ export class IntersolveVisaApiService {
     return timeLeftBeforeExpire > 60000;
   }
 
-  public async createCustomer(
-    payload: IntersolveCreateCustomerDto,
-  ): Promise<IntersolveCreateCustomerResponseBodyDto> {
+  public async createCustomer(input: {
```

Comment:
> The lack of destructuring in the arguments of the functions in this file seems inconsistent (ie. in some functions the parameters are destructured, but in others they are not.

Probably some becomes clear by reading on how we thought about this:
https://github.com/global-121/121-platform/wiki/Guidelines-Software-Architecture#c-function-signatures-in-services

> For consistency's sake, and to avoid having to repeat the input.PARAM, it's better to restructure. At times, we avoid destructuring to be able to re-use the input directly. We can solve that with this pattern:

Also see this code that we already talked about in a previous review:
https://github.com/global-121/121-platform/blob/177514dca1820f06d50889d057a7f33ddf7c0cb2/services/121-service/src/payments/fsp-integration/intersolve-visa/intersolve-visa.service.ts#L68

where we said that having "input" is preferable for long functions.

Let's have a conversation!

Reply:
> Probably some becomes clear by reading on how we thought about this:
https://github.com/global-121/121-platform/wiki/Guidelines-Software-Architecture#c-function-signatures-in-services

AFAIU this only talks about how/where to declare the types, not necessarily how to consume/declare the data itself.

> Also see this code that we already talked about in a previous review:

I don't recall this 😄 

> Let's have a conversation!

Sure thing, hit me up when you prefer (maybe once we've resolved everything that's easily resolvable, and are left only with things that need discussion).

Reply:
See changed doTransferOrIssueCard() for how to do this when the interface is defined in a separate file.